### PR TITLE
[OAuth] Up access and refresh tokens' length

### LINKED ISSUE #11608: Database field too small for OAuth access token when using Keycloak as identity provider
**Sylius version affected**: 1.7.6

**Description**  
Trying to add users via OAuth (using hwi/HWIOAuthBundle) fails:
`SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'access_token' at row 1 at /var/www/sylius/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:129, PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'access_token' at row 1 at /var/www/sylius/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:127)`
Affected identity provider: Keycloak

**Steps to reproduce**  
- install Sylius following the cookbook
- `composer require hwi/HWIOAuthBundle`
- setup identity provider following the documentation

**Possible Solution**  
Access tokens can be of very different sizes, depending on contained data.
Suggestion: 
`ALTER TABLE sylius_user_oauth MODIFY access_token VARCHAR(8192);`  
`ALTER TABLE sylius_user_oauth MODIFY refresh_token VARCHAR(8192);`

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | fixes #11608 #13505, replaces #14072                      |
| License         | MIT                                                          |

### CODE REVIEW

**Review by jakubtobiasz — APPROVED** [review]

**Review by coldic3 — APPROVED** [review]

**Review by Rafikooo — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Migrations/Version20230327121633.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+final class Version20230327121633 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Allow longer access and refresh tokens for OAuth users';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE sylius_user_oauth CHANGE access_token access_token VARCHAR(8192) DEFAULT NULL, CHANGE refresh_token refresh_token VARCHAR(8192) DEFAULT NULL');
```
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/Migrations/Version20230327121633.php:28]: According to:
https://github.com/Sylius/Sylius/issues/13505#issuecomment-1015754276

Shouldn't we resize the column to an even larger size?

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :tada:

File: src/Sylius/Bundle/CoreBundle/Migrations/Version20230327121633.php (hunk 1 above)
Comment (Nek-) [inline src/Sylius/Bundle/CoreBundle/Migrations/Version20230327121633.php:28]: For your information, 8192 will not work with MariaDB and innoDB. https://mariadb.com/kb/en/troubleshooting-row-size-too-large-errors-with-innodb/

Comment (NoResponseMate) [reply src/Sylius/Bundle/CoreBundle/Migrations/Version20230327121633.php:28]: Think this has been already addressed in #14930, the update changed the field type from varchar simply to text.

**Pull request merged**